scylla-cql/Cargo.toml

```diff
@@ -21,6 +21,7 @@ uuid = "1.0"
 thiserror = "1.0"
 bigdecimal = "0.2.0"
 num-bigint-03 = { package = "num-bigint", version = "0.3" }
```

Comment:
Apart from the `CqlValueDisplayer`, what else in the driver currently requires this crate?

Reply:
The constructors of `CqlVarint` make use of the crate. The reason is that the `num_bigint` library normalizes the user provided `u8` slice/array.

For example:
```rust
pub fn from_signed_bytes_be(digits: &[u8]) -> Self {
        let normalized_bigint = num_bigint_03::BigInt::from_signed_bytes_be(digits);
        Self(normalized_bigint.to_signed_bytes_be())
}
```

I wasn't sure if this is necessary as Scylla accepts non-normalized bytes (with leading zeros) and stores them as is (result of the query on the inserted data has leading zeros as well).

The only issue is a following client-side scenario:
```rust
let one_with_leading_zeros_be = vec![0x00, 0x00, 0x00, 0x01];
let cql_varint: CqlVarint = CqlVarint::from_signed_bytes_be(&one_with_leading_zeros_be);
let bigint: num_bigint::BigInt = cql_varint.clone().into();
let converted_cql_varint: CqlVarint = bigint.into();

// This panics!
// cql_varint.signed_bytes_be().len() == 4
// converted_cql_varint.to_signed_bytes_be().len() == 1
assert_eq!(cql_varint.to_signed_bytes_be(), converted_cql_varint.to_signed_bytes_be());
```

If we want the dependency to be optional, I see two ways to achieve it:
1. ignore above issue and insert non-normalized data to DB
2. reimplement normalization algorithm in the driver

Reply:
I don't think that lack of normalization should be a big issue. Converting the CqlVarint to a useful form (i.e. BigInt) will normalize it. Besides, in order to create a non-normalized value one has to create the CqlVarint from bytes - I think this is rare.